specs/Deployment/MSIXPackages.md

```diff
@@ -0,0 +1,189 @@
+processes from the existing Main package to be terminated.
+
+For these reasons developers are strongly encouraged to only rely on the Main package when
+necessary.
+
+If a process from the Main package is necessary, it should be short-lived or support a save/restore
+execution model. The former shrinks the possibility a running process when a servicing event
+occurs. The latter ensures if Main package processes are running when a servicing event occurs, the
+running processes can be terminated (e.g. via [DeploymentOptions.ForceTargetApplicationShutdown](https://docs.microsoft.com/uwp/api/windows.management.deployment.deploymentoptions))
+to allow the servicing event to occur and restarted upon its completion, with only a transient (and
+possibly unnoticed) interruption of service.
+
+Further, if/when the Main package is necessary, developers are encouraged to put as
+little as practical in the Main package and use the Framework package as much as possible. For
+example, if a StartupTask is needed you can declare the StartupTask in the Main package's
+`appxmanifest.xml` and provide an executable with a trivial 1% implementation e.g.
+
+```c
+int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, PWSTR pCmdLine, int nCmdShow)
+{
+    return MyFeatureWinMain(hInstance, hPrevInstance, pCmdLine, nCmdShow);
+}
+```
+
+and export `MyFeatureWinMain` from a DLL in the Framework package to get the implementation's
+other 99%.
+
+### Role - Singleton
+
+The Singleton package supplements the Main package, for when functionality is needed by a single
+process spanning all versions of Windows App SDK.
+
+### Dynamic Dependency Lifetime Manager (DDLM)
+
+The [Dynamic Dependency API](https://github.com/microsoft/WindowsAppSDK/blob/main/specs/dynamicdependencies/DynamicDependencies.md)
+resolves a package dependency to a specific framework package and make it available for a process,
+but Windows doesn't know the package is in use by this process. Lacking this "package in use"
+knowledge, Windows could service the package (remove, remediate, repair, ...) in negatively
+impactful ways. To prevent this (without servicing Windows) we need...
+
+1. ...a running process running with package identity...
+2. ...with an `appxmanifest.xml` declaring a `<PackageDependnecy>` on the framework package...
+3. ...running for as long as the framework package is used by the app.
+
+This is accomplished for the Microsoft.WindowsAppRuntime framework package via a Packaged COM
+out-of-process (OOP) server in a DDLM package, selected and managed via the Bootstrap API. See the
+[Dynamic Dependency spec](https://github.com/microsoft/WindowsAppSDK/blob/main/specs/dynamicdependencies/DynamicDependencies.md)
+for more details.
+
+## Naming
+
+Windows App SDK's MSIX packages use the following naming rules for package identity:
+
+* Name = Microsoft.WindowsAppRuntime[.SubName].<Major>.<Minor>[-VersionTag]
```

Comment:
Version tags are missing when the Markdown is rendered.

```suggestion
* Name = Microsoft.WindowsAppRuntime[.SubName].\<Major>.\<Minor>[-VersionTag]
```

Reply:
Gah! Thanks!